langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/Round.java

```diff
@@ -31,7 +34,46 @@
 //)
 public class Round {
 
-    public static double round(double x) {
-        return Math.rint(x);
+    public static double round(double x, long fractionDigits) {
+        if (isSpecialValue(x)) {
+            return x;
+        }
+        if (fractionDigits == 0) {
+            return Math.rint(x);
+        }
+        if (fractionDigits > Integer.MAX_VALUE) {
+            // As per IEEE, exponent of double value lies from -308 to 307.
+            // Also, the maximum decimal digits that can have in double value are 15 or 16.
+            // Therefore, if `fractionDigits` is very large, `x` will not be changed.
+            return x;
+        }
+        if (fractionDigits < Integer.MIN_VALUE) {
+            return 0;
+        }
+        // Down cast can be done safely because of above condition.
+        int fractionDigitsAsInt = (int) fractionDigits;
+        BigDecimal xInBigDecimal = BigDecimal.valueOf(x);
+        BigDecimal xTmp = xInBigDecimal;
+        int digitsTmp = fractionDigitsAsInt;
+        if (digitsTmp > 0) {
+            while (digitsTmp-- > 0) {
```

Comment:
Shall we add some comments to explain this logic in high level. Looks to me like if we multiply by 100 this will go faster.
BigDecimal has a `scale` method to know decimal points in advance. So we can check if 
`x.scale() <= digitsTmp` in one line